sdk/src/main/java/io/customer/sdk/CustomerIO.kt

```diff
@@ -167,20 +167,72 @@ class CustomerIO internal constructor(
     ) {
         private val sharedInstance = CustomerIOShared.instance()
         private var client: Client = Client.Android(Version.version)
-        private var timeout = AnalyticsConstants.HTTP_REQUEST_TIMEOUT
+        private var timeout = CustomerIOConfig.Companion.AnalyticsConstants.HTTP_REQUEST_TIMEOUT
         private var shouldAutoRecordScreenViews: Boolean =
-            AnalyticsConstants.SHOULD_AUTO_RECORD_SCREEN_VIEWS
+            CustomerIOConfig.Companion.AnalyticsConstants.SHOULD_AUTO_RECORD_SCREEN_VIEWS
         private var autoTrackDeviceAttributes: Boolean =
-            AnalyticsConstants.AUTO_TRACK_DEVICE_ATTRIBUTES
+            CustomerIOConfig.Companion.AnalyticsConstants.AUTO_TRACK_DEVICE_ATTRIBUTES
         private val modules: MutableMap<String, CustomerIOModule<out CustomerIOModuleConfig>> =
             mutableMapOf()
-        private var logLevel: CioLogLevel = SDKConstants.LOG_LEVEL_DEFAULT
+        private var logLevel: CioLogLevel =
+            CustomerIOConfig.Companion.SDKConstants.LOG_LEVEL_DEFAULT
         internal var overrideDiGraph: CustomerIOComponent? = null // set for automated tests
         private var trackingApiUrl: String? = null
         private var backgroundQueueMinNumberOfTasks: Int =
-            AnalyticsConstants.BACKGROUND_QUEUE_MIN_NUMBER_OF_TASKS
+            CustomerIOConfig.Companion.AnalyticsConstants.BACKGROUND_QUEUE_MIN_NUMBER_OF_TASKS
         private var backgroundQueueSecondsDelay: Double =
-            AnalyticsConstants.BACKGROUND_QUEUE_SECONDS_DELAY
+            CustomerIOConfig.Companion.AnalyticsConstants.BACKGROUND_QUEUE_SECONDS_DELAY
+
+        constructor(
+            siteId: String,
+            apiKey: String,
+            region: Region = Region.US,
+            appContext: Application,
+            config: Map<String, Any?>
+        ) : this(siteId, apiKey, region, appContext) {
+            setupConfig(config)
+        }
```

Comment:
Since we have `null` check in `setupConfig` and `JvmOverloads` on default constructor, we could simply add `config: Map<String, Any?>? = null` to default constructor?

Or update `setupConfig` to `if (config == null || config.isEmpty()) return this` and default constructor to `config: Map<String, Any?> = emptyMap()`? But in either case, I think we can skip this constructor.

Reply:
Absolutely, not needed, created a separate secondary just we keep primary as recommended to the world and use secondary just internally. Don't want people to get confused by seeing this parameter and sending maps of unrelated stuff.

Reply:
Makes sense to me. Adding this idea as comment might help so anyone looking into it in future doesn't get the same confusion.